manifests/server/pg_hba_rule.pp

```diff
@@ -12,10 +12,12 @@
 
   # Needed for testing primarily, support for multiple files is not really
   # working.
-  $target      = $postgresql::server::pg_hba_conf_path
+  $target             = $postgresql::server::pg_hba_conf_path,
+  $postgresql_version = $postgresql::server::_version,
+  $manage_pg_hba_conf = $postgresql::server::manage_pg_hba_conf
```

Comment:
This will create a parse-time order dependency between this code and `postgresql::server`. If for some reason the puppet compiler evaluates a `postgresql::server::pg_hba_rule` before encountering and evaluating the `postgresql::server` inclusion, this will pick up the wrong defaults here.